package/MDAnalysis/coordinates/TRJ.py

```diff
@@ -640,7 +643,10 @@ def _read_frame(self, frame):
                 self.n_frames))
         # note: self.trjfile.variables['coordinates'].shape == (frames, n_atoms, 3)
         ts._pos[:] = self._get_var_and_scale('coordinates', frame)
-        ts.time = self._get_var_and_scale('time', frame)
+        try:
+            ts.time = self._get_var_and_scale('time', frame)
+        except KeyError:
+            ts.time = -1
```

Comment:
I'm going to ping @MDAnalysis/coredevs for ideas on this one. Technically this case should only happen for non-temporal trajectory data, i.e. "a bunch of static frames that aren't related to each other by time".

Should we either a) use the timestep default, not set time and let it be set to dt * frame + offset, or b) set it to -1, c) some other standard we've been using somewhere that I've not kept up with.

I originally advocated for b), but look at the PDB reader I'm starting to convince myself that a) might be the right approach (since PDB trajectories are also meant to be frame not linked via time).

Reply:
Ah sorry, I should have probably clarified, the default action for `ts.time` is to do `dt * frame + offset` so I believe the answer is just not to set `ts.time` at all if the time variable doesn't exist (I _think_, it needs a bit of digging around the base and timestep files and I'm a bit short on time to be 100% sure today sorry!).

I _think_ the chainreader should overwrite `frames` at that level, although I know there were some issues opened about its behaviour.

Reply:
I tried as you suggested and it worked as intended, thanks!